File: src/main/java/wraith/fabricaeexnihilo/modules/ModEffects.java
package wraith.fabricaeexnihilo.modules;

import net.minecraft.entity.effect.StatusEffect;
import net.minecraft.entity.effect.StatusEffectCategory;
import net.minecraft.registry.Registries;
import net.minecraft.registry.Registry;
import wraith.fabricaeexnihilo.modules.witchwater.WitchWaterStatusEffect;
import wraith.fabricaeexnihilo.util.Color;

import static wraith.fabricaeexnihilo.FabricaeExNihilo.id;

public final class ModEffects {

    public static final StatusEffect MILKED = new StatusEffect(StatusEffectCategory.NEUTRAL, Color.WHITE.toInt()) {
    };
    /**
     * A status effect used to mark entities that have been processed by the witch water code so that they are no longer
     * processed.
     */
    public static final WitchWaterStatusEffect WITCH_WATERED = new WitchWaterStatusEffect();

    public static void registerEffects() {
        Registry.register(Registries.STATUS_EFFECT, id("witch_watered"), WITCH_WATERED);
        Registry.register(Registries.STATUS_EFFECT, id("milked"), MILKED);
    }

}

File: src/main/kotlin/exnihilofabrico/modules/ModEffects.kt
package exnihilofabrico.modules

import exnihilofabrico.id
import exnihilofabrico.modules.witchwater.WitchWaterStatusEffect
import exnihilofabrico.util.Color
import net.minecraft.entity.effect.StatusEffect
import net.minecraft.entity.effect.StatusEffectType
import net.minecraft.util.registry.Registry

object ModEffects {
    /**
     * A status effect used to mark entities that have been processed by the witch water code so that they are no longer processed.
     */
    val WITCHWATERED: StatusEffect = WitchWaterStatusEffect
    val MILKED = object: StatusEffect(StatusEffectType.NEUTRAL, Color.WHITE.toInt()) {}

    fun registerEffects(registry: Registry<StatusEffect>) {
        Registry.register(registry, id("witchwatered"), WITCHWATERED)
        Registry.register(registry, id("milked"), MILKED)
    }
}